From d6e8359a824f08c35f2ea11c17c3c623220525d5 Mon Sep 17 00:00:00 2001
From: Steven Barrett <steven@liquorix.net>
Date: Mon, 20 Sep 2021 20:18:14 -0500
Subject: [PATCH 1/2] sched/alt: Fix cgroup time accounting

As fixed by Alfred in the link below:
https://gitlab.com/alfredchen/linux-prjc/-/issues/40#note_682574650

Invoke "cgroup_account_cputime" in "update_curr".
---
 kernel/sched/alt_core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 7c998e6..258f11b 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -3468,6 +3468,7 @@ static inline void update_curr(struct rq *rq, struct task_struct *p)
 	s64 ns = rq->clock_task - p->last_ran;
 
 	p->sched_time += ns;
+	cgroup_account_cputime(p, ns);
 	account_group_exec_runtime(p, ns);
 
 	p->time_slice -= ns;
-- 
2.33.0.328.g8b7c11b866

